x-pack/plugins/observability_solution/observability/server/services/slo/transform_generators/synthetics_availability.ts

```diff
@@ -0,0 +1,193 @@
+    // we'll output documents for each environment value
+    const extraGroupByFields = {
+      'monitor.id': { terms: { field: 'monitor.id' } },
+      'observer.name': { terms: { field: 'observer.name' } },
+      config_id: { terms: { field: 'config_id' } },
+      ...(hasTags && {
+        tags: { terms: { field: 'tags' } },
+      }),
+      ...(hasProjects && {
+        'monitor.project.id': { terms: { field: 'monitor.project.id' } },
+      }),
+    };
+
+    // These are the group by fields that will be used in `groupings` key
+    // in the summary and rollup documents. For Synthetics, we want to use the
+    // user-readible `monitor.name` and `observer.geo.name` fields by default,
+    // unless otherwise specified by the user.
+    const flattenedGroupBy = [slo.groupBy].flat().filter((value) => !!value);
+    const groupings =
+      flattenedGroupBy.length && !flattenedGroupBy.includes(ALL_VALUE)
+        ? slo.groupBy
+        : ['monitor.name', 'observer.geo.name'];
+
+    return this.buildCommonGroupBy(
+      { ...slo, groupBy: groupings },
+      '@timestamp',
+      extraGroupByFields
+    );
+  }
+
+  private buildSource(slo: SLO, indicator: SyntheticsAvailabilityIndicator, spaceId: string) {
+    const queryFilter: estypes.QueryDslQueryContainer[] = [
+      { term: { 'summary.final_attempt': true } },
+      { term: { 'meta.space_id': spaceId } },
+      {
+        range: {
+          '@timestamp': {
+            gte: `now-${slo.timeWindow.duration.format()}/d`,
+          },
+        },
+      },
+    ];
+    const { monitorIds, tags, projects } = buildParamValues({
+      monitorIds: indicator.params.monitorIds || [],
+      tags: indicator.params.tags || [],
+      projects: indicator.params.projects || [],
+    });
+
+    if (!monitorIds.includes(ALL_VALUE) && monitorIds.length) {
+      queryFilter.push({
+        terms: {
+          'monitor.id': monitorIds,
+        },
+      });
+    }
+
+    if (!tags.includes(ALL_VALUE) && tags.length) {
+      queryFilter.push({
+        terms: {
+          tags,
+        },
+      });
+    }
+
+    if (!projects.includes(ALL_VALUE) && projects.length) {
+      queryFilter.push({
+        terms: {
+          'monitor.project.id': projects,
+        },
+      });
+    }
+
+    if (!!indicator.params.filter) {
+      queryFilter.push(getElasticsearchQueryOrThrow(indicator.params.filter));
+    }
+
+    return {
+      index: 'synthetics-*',
+      runtime_mappings: {
+        ...this.buildCommonRuntimeMappings(slo),
+      },
+      query: {
+        bool: {
+          filter: queryFilter,
+        },
+      },
+    };
+  }
+
+  private buildDestination() {
+    return {
+      pipeline: SLO_INGEST_PIPELINE_NAME,
+      index: SLO_DESTINATION_INDEX_NAME,
+    };
+  }
+
+  private buildAggregations(slo: SLO) {
+    return {
+      'slo.numerator': {
+        filter: {
+          range: {
+            'summary.up': {
+              gte: 1,
+            },
+          },
```

Comment:
```suggestion
          term: {
            'monitor.status': 'up',
          },
```

Reply:
What's the difference between `summary.up > 1` and `monitor.status: "up"`? Is `summary.up` accounting for retries, and `monitor.status` the final status?

Reply:
Okay we're gonna use `monitor.status`. Confirmed.